### PR TITLE
Validate repo and tag names in retag

### PR DESCRIPTION
**Pull request by cd1989**
Signed-off-by: cd1989 <chende@caicloud.io>

Related to issue: https://github.com/goharbor/harbor/issues/6540

### CODE REVIEW

File: src/common/utils/validate.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	"regexp"
+)
+
+const nameComponent = `[a-z0-9]+((?:[._]|__|[-]*)[a-z0-9]+)*`
+
+// TagRegexp is regular expression to match image tags, for example, 'v1.0'
+var TagRegexp = regexp.MustCompile(`^[\w][\w.-]{0,127}$`)
+
+// RepoRegexp is regular expression to match repo name, for example, 'busybox', 'stage/busybox'
+var RepoRegexp = regexp.MustCompile(fmt.Sprintf("^%s(/%s)*$", nameComponent, nameComponent))
```
Comment (wy65701436) [inline src/common/utils/validate.go]: according to the docker official document, https://forums.docker.com/t/docker-registry-v2-spec-and-repository-naming-rule/5466

 If a repository name has two or more path components, they must be separated by a forward slash ("/").

Why do we only support two path components?

File: src/core/api/repository.go [hunk 2]
```diff
@@ -463,6 +463,16 @@ func (ra *RepositoryAPI) Retag() {
 		return
 	}
 
+	if !utils.ValidateRepo(repo) {
```
Comment (wy65701436) [inline src/core/api/repository.go]: move the validation up to line 437, it should be the first action in the API.

File: src/common/utils/validate.go (hunk 1 above)
Comment (cd1989) [reply src/common/utils/validate.go]: You may misunderstand it, it do support multiple components.

File: src/core/api/repository.go (hunk 2 above)
Comment (cd1989) [reply src/core/api/repository.go]: done

**Pull request merged**